pandas/core/generic.py

```diff
@@ -5601,53 +5601,95 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
+            Original input with those values above/below the
+            `upper`/`'lower` thresholds set to the threshold values.
+
+        See Also
+        --------
+        DataFrame.clip : Trim values at input threshold(s).
+        Series.clip : Trim values at input threshold(s).
+        Series.clip_lower : Return copy of the input with values below given
+            value(s) truncated.
+        Series.clip_upper : Return copy of input with values above given
+            value(s) truncated.
+        DataFrame.clip_lower : Return copy of the input with values below given
+            value(s) truncated.
+        DataFrame.clip_upper : Return copy of input with values above given
+            value(s) truncated.
+        DataFrame.quantile : Return values at the given quantile over requested
+            axis, a la numpy.percentile.
 
         Examples
         --------
+        >>> some_data = {'a': [-1, -2, -100], 'b': [1, 2, 100]}
+        >>> df = pd.DataFrame(some_data, index = ['foo', 'bar', 'foobar'])
         >>> df
-                  0         1
-        0  0.335232 -1.256177
-        1 -1.367855  0.746646
-        2  0.027753 -1.176076
-        3  0.230930 -0.679613
-        4  1.261967  0.570967
-
-        >>> df.clip(-1.0, 0.5)
-                  0         1
-        0  0.335232 -1.000000
-        1 -1.000000  0.500000
-        2  0.027753 -1.000000
-        3  0.230930 -0.679613
-        4  0.500000  0.500000
-
-        >>> t
-        0   -0.3
-        1   -0.2
-        2   -0.1
-        3    0.0
-        4    0.1
-        dtype: float64
-
-        >>> df.clip(t, t + 1, axis=0)
-                  0         1
-        0  0.335232 -0.300000
-        1 -0.200000  0.746646
-        2  0.027753 -0.100000
-        3  0.230930  0.000000
-        4  1.100000  0.570967
+            a   b
+        foo -1  1
+        bar -2  2
+        foobar  -100    100
+
+        >>> df.clip(lower=-10, upper=10)
+            a   b
+        foo -1  1
+        bar -2  2
+        foobar  -10 10
+
+        You can clip each column or row with different thresholds by passing
+        a ``Series`` to the lower/upper argument. Use the axis argument to clip
+        by column or rows.
+
+        >>> col_thresh = pd.Series({'a': -5, 'b': 5})
+        >>> df.clip(lower=col_thresh, axis='columns')
+            a   b
+        foo -1  5
+        bar -2  5
+        foobar  -5  100
+
+        Clip the foo, bar, and foobar rows with lower thresholds 5, 7, and 10.
+
+        >>> row_thresh = pd.Series({'foo': 0, 'bar': 1, 'foobar': 10})
+        >>> df.clip(lower=row_thresh, axis='index')
+            a   b
+        foo 0   1
+        bar 1   2
+        foobar  10  100
+
+        `Winsorizing <https://en.wikipedia.org/wiki/Winsorizing>`__ is a
+        related method, whereby the data are clipped at
+        the 5th and 95th percentiles.
+
+        >>> lower, upper = df.quantile(0.05), df.quantile(0.95)
+        >>> df.clip(lower=lower, upper=upper, axis='columns')
```

Comment:
Not sure if it's just because I'm too tired, but is `axis` useful here? If the threshold is a scalar, doesn't seem to have an effect, right?

Reply:
Axis is required here since `df.quantile` returns a series with the quantiles for each column.  Running without the axis argument results in an error.

Reply:
I'd find useful having this comment in the explanations before the test. :)